### ISSUE TITLE
Error returned from spectral_smooth in Python 3.7.0

### ISSUE THREAD

**Issue by ashleythomasbarnes** [radio-astro-tools/spectral-cube#514]
I am trying to run some code I previously had working in Python 2.7.0 that used spectral-cube to do spectral smoothing (the code I am using is almost identical to that shown in the smoothing example: https://spectral-cube.readthedocs.io/en/latest/smoothing.html). However, I get the following error: 

```
AttributeError                            Traceback (most recent call last)
<ipython-input-5-db1df68caa65> in <module>()
     26 
     27 kernel = Gaussian1DKernel(gaussian_width)
---> 28 smcube = cube.spectral_smooth(kernel)

/anaconda3/lib/python3.7/site-packages/spectral_cube/spectral_cube.py in spectral_smooth(self, kernel, convolve, verbose, use_memmap, num_cores, **kwargs)
   2768                                                       use_memmap=use_memmap,
   2769                                                       verbose=verbose,
-> 2770                                                       **kwargs)
   2771 
   2772     def spectral_interpolate(self, spectral_grid,

/anaconda3/lib/python3.7/site-packages/spectral_cube/spectral_cube.py in _apply_function_parallel_spectral(self, function, data, num_cores, verbose, use_memmap, parallel, **kwargs)
   2719 
   2720             for arg in spectra:
-> 2721                 _apply_function(arg, outcube, function, **kwargs)
   2722                 pbu()
   2723 

/anaconda3/lib/python3.7/site-packages/spectral_cube/spectral_cube.py in _apply_function(arguments, outcube, function, **kwargs)
     98 
     99     if any(includemask):
--> 100         outcube[:,jj,ii] = function(spec, **kwargs)
    101     else:
    102         outcube[:,jj,ii] = spec

/anaconda3/lib/python3.7/site-packages/astropy/utils/decorators.py in convolve(array, kernel, boundary, fill_value, nan_treatment, normalize_kernel, mask, preserve_nan, normalization_zero_tol)
    840             name = func.__name__
    841 
--> 842         func = make_function_with_signature(func, name=name, **wrapped_args)
    843         func = functools.update_wrapper(func, wrapped, assigned=assigned,
    844                                         updated=updated)

/anaconda3/lib/python3.7/site-packages/astropy/nddata/decorators.py in wrapper(data, *args, **kwargs)
    243                                   AstropyUserWarning)
    244 
--> 245             result = func(data, *args, **kwargs)
    246 
    247             if unpack and repack:

/anaconda3/lib/python3.7/site-packages/astropy/convolution/convolve.py in convolve(array, kernel, boundary, fill_value, nan_treatment, normalize_kernel, mask, preserve_nan, normalization_zero_tol)
    336     if normalize_kernel:
    337         if not nan_interpolate:
--> 338             result /= kernel_sum
    339     else:
    340         if nan_interpolate:

/anaconda3/lib/python3.7/site-packages/astropy/units/quantity.py in __array_ufunc__(self, function, method, *inputs, **kwargs)
    468             return result
    469 
--> 470         return self._result_as_quantity(result, unit, out)
    471 
    472     def _result_as_quantity(self, result, unit, out):

/anaconda3/lib/python3.7/site-packages/astropy/units/quantity.py in _result_as_quantity(self, result, unit, out)
    507         # the output is of the correct Quantity subclass, as it was passed
    508         # through check_output.
--> 509         out._set_unit(unit)
    510         return out
    511 

AttributeError: 'numpy.ndarray' object has no attribute '_set_unit'

```

I am using:
```
astropy.__version__ = '3.1'
numpy.__version__ = '1.15.4'
spectral_cube.__version__ = '0.4.3'
```

**Comment by keflavich** [comment]
I can't reproduce this issue on my current install, but I'll test with your exact configuration.  Could you post the FITS header of the file you were trying to smooth, though?  I'm curious whether it has something to do with that.

**Comment by ashleythomasbarnes** [comment]
I have just tried this on several datasets and I get the same error. An example cube I am using can be found here: 

[https://www.dropbox.com/s/iinr74k133r7w15/testing.fits?dl=0](url)

which has the following information: 

```
SpectralCube with shape=(34, 50, 50) and unit=K:
 n_x:     50  type_x: RA---TAN  unit_x: deg    range:   287.683161 deg:  287.821004 deg
 n_y:     50  type_y: DEC--TAN  unit_y: deg    range:     8.910102 deg:    9.046256 deg
 n_s:     34  type_s: VRAD      unit_s: km / s  range:      -49.800 km / s:     -30.000 km / s
```

```
WCS Keywords

Number of WCS axes: 3
CTYPE : 'RA---TAN'  'DEC--TAN'  'VRAD'  
CRVAL : 287.6353625  9.093481111111  0.0  
CRPIX : 67.0  67.0  84.0  
PC1_1 PC1_2 PC1_3  : 1.0  0.0  0.0  
PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0  
PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0  
CDELT : -0.0027777777777778  0.0027777777777778  600.0000238419  
NAXIS : 50  50  34
```

The code I am using to smooth is the following: 

```
import numpy as np
import astropy.units as u
from spectral_cube import SpectralCube
from astropy.convolution import Gaussian1DKernel


spectralsmoothfactor = 2
inputfile = 'testing.fits'

cube = SpectralCube.read(inputfile, format = 'fits', unit = u.K, beam_threshold = 0.05)
cube = cube.with_spectral_unit(u.km / u.s)

cube.beam_threshold = 0.5

current_resolution = cube.header['CDELT3']
current_resolution = current_resolution * u.km/u.s

target_resolution = current_resolution * spectralsmoothfactor
fwhm_factor = np.sqrt(8*np.log(2))
current_pixel_scale = current_resolution
gaussian_width = ((target_resolution**2 - current_resolution**2)**0.5 / current_pixel_scale / fwhm_factor)

kernel = Gaussian1DKernel(gaussian_width)
smcube = cube.spectral_smooth(kernel)
```

**Comment by keflavich** [comment]
Thanks, I'll investigate today...

**Comment by e-koch** [comment]
It's coming from passing `gaussian_width` as a Quantity to the kernel. This fixes it:
```
kernel = Gaussian1DKernel(gaussian_width.value)
smcube = cube.spectral_smooth(kernel)
```
@keflavich - this might be fixed with #507 if the spectra passed to the convolution function are unitless Quantities.
